lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/context/ContextRepresenters.java

```diff
@@ -53,8 +56,22 @@
   @Inject
   private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
                               final ContextFactory contextFactory) {
+    this(messageDispatcher, contextFactory, Optional.<DriverRestartManager>empty());
+  }
+
+  @Inject
+  private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
+                              final ContextFactory contextFactory,
+                              final DriverRestartManager driverRestartManager) {
+    this(messageDispatcher, contextFactory, Optional.of(driverRestartManager));
+  }
+
+  private ContextRepresenters(final EvaluatorMessageDispatcher messageDispatcher,
+                              final ContextFactory contextFactory,
+                              final Optional<DriverRestartManager> driverRestartManager) {
```

Comment:
I'll file a separate item for this. Doing this will change other portions of the code not relevant to the aim of this PR.